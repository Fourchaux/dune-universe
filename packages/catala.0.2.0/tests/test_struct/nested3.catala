@Article@

/*
new struct S:
  data x content int
  data y content bool

new struct T:
  data a content S
  data b content S 

new scope A:
  param t content T 

new scope B:
  param t content T 
  param a scope A 
  param out content int

scope A:
  def t := T {
    -- a : S {
      -- x : 0
      -- y : false 
    }
    -- b : S {
      -- x : 1
      -- y : true
    }
  }  

scope B:
  def t := a.t 
  def out := if t.a.y then t.a.x else (if t.b.y then t.b.x else 42)
*/