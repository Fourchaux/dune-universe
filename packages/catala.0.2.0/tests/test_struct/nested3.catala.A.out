[RESULT] Computation successful! Results:
[RESULT] t = ("a": ("x": 0,"y": false),"b": ("x": 1,"y": true))
